### PR TITLE
Fix classes dirs regression

### PR DESCRIPTION
**Pull request by jvandort**
Introduced here:
https://github.com/gradle/gradle/pull/21543/files#diff-d50000c56d36151bbac24c48dd8259210337e8707ccf19df58be08ec7905825eR195-R201

### CODE REVIEW

**Comment by jvandort** [comment]
@bot-gradle test QFL

**Comment by jvandort** [comment]
@bot-gradle test qfl

**Comment by jvandort** [comment]
@bot-gradle test this

**Comment by jvandort** [comment]
@bot-gradle test this

File: subprojects/ear/src/test/groovy/org/gradle/plugins/ear/EarPluginTest.groovy [hunk 1]
```diff
@@ -103,7 +103,7 @@ class EarPluginTest extends AbstractProjectBuilderSpec {
 
         then:
         task instanceof Ear
-        task dependsOn(JavaPlugin.CLASSES_TASK_NAME)
+        task dependsOn(JavaPlugin.CLASSES_TASK_NAME, JavaPlugin.COMPILE_JAVA_TASK_NAME)
```
Comment (ljacomet) [inline subprojects/ear/src/test/groovy/org/gradle/plugins/ear/EarPluginTest.groovy:106]: What is the exact meaning of having to change these assertions?
In practice, the logical dependency chain should be ear -> classes -> compileJava
With the change, it seems like we flatten some transitive dependencies one level up.

**Comment by ljacomet** [comment]
@bot-gradle test RfR

**Review by big-guy — APPROVED** [review]

**Comment by jvandort** [comment]
@bot-gradle test and merge

**Pull request merged**